x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/direct/DirectBlobContainerIndexInput.java

```diff
@@ -190,10 +194,38 @@ private int readFromNewSequentialStream(int part, long pos, byte[] b, int offset
             return 0;
         }
 
+        final long startTimeNanos = directory.statsCurrentTimeNanos();
+
         // if we open a stream of length streamLength then it will not be completely consumed by this read, so it is worthwhile to open
         // it and keep it open for future reads
         final InputStream inputStream = openBlobStream(part, pos, streamLength);
-        streamForSequentialReads = new StreamForSequentialReads(inputStream, part, pos, streamLength);
+        streamForSequentialReads = new StreamForSequentialReads(new FilterInputStream(inputStream) {
```

Comment:
This is were "optimized_read_bytes" are captured for direct index inputs. Note that they are only added to stats when the stream is closed. This is useful in case the inner input stream reads all remaining bytes on closing (though there is no test for this).